### DISCUSSION TITLE
How to get the AI ​​to initiate a conversation itself

### DISCUSSION THREAD

**Discussion by Billyfury31** [oobabooga/text-generation-webui#3848] [Q&A] [answered]
As the title suggests, I would like the AI ​​to start a conversation on its own after a while, this will give it a little more life. Problem, otherwise it wouldn't be fun x), I don't know how to do it lol, apart from sending a message via the API automatically but that would be a bit of a cheat. As a last resort, if I can't find a solution, I will use this method. I tried to do my own research but without success, unless I'm looking wrong but I don't think so.
`I use the Wizard-Vicuna-13B-Uncensored-GPTQ model on google colab.`
In short, if you could help me, if you knew the answer or if you said to yourself "But it's stupid, it's impossible" xD Well tell me, even if you just had a little information to guide me, that's it. welcome, thank you in advance for your response.
(If my English is weird, I use Google Translate, in French)

**Comment by VenkatLohithDasari** [comment]
You simply cannot, These AI Models work on the basis of Input and output, You provide some input, and it will predict the output.

The only possible method would be using a python script which sends initial input, asking AI to talk to itself via providing the first question to itself, and then the script will send back that question to AI, asking AI to send an answer as well as the next question then Script will pickup that next question from the output and then sends back to AI as Input once again. this cycle will repeat itself forever.

**Reply by Billyfury31** [reply]
First of all, thank you for your quick response. The fact that the AI ​​can't say things spontaneously is due to the model I use or is it impossible with all the models at the moment?

**Reply by VenkatLohithDasari** [reply, marked as answer]
It's not a limitation of any specific AI model; rather, it's a fundamental characteristic of how AI works. 

Quick Answer - It's impossible with all models. I should emphasize, that it's impossible with our current state of AI. You have to wait for True AGI.

Let's make it much clearer, Imagine AI models as incredibly complex math equations. They don't have thoughts or self-awareness like humans. Instead, they process input (your prompt) and generate output based on these mathematical functions. It's similar to a simple math equation like 'f(x) = 2x' which doubles the input: if you give it 'x=2', it'll give you '4' as the result. In AI, this equation is represented by the model.

When you interact with an AI model, you provide input like 'Hello, AI.' The model processes this input using its intricate math functions and gives you an output, like 'Hello! How can I assist you today?' So, AI doesn't think or talk to itself independently because it's always reacting to the input it receives.

Think of it like asking a calculator to do math for you. It only responds when you input numbers and operations, they're tools that respond to the input they're given.

**Reply by Billyfury31** [reply]
Yeah, don't worry, I know more or less that a model only includes numbers, I just seemed a bit stupid in my previous messages. I suspected a bit of the answer, it is mandatory to give a prompt to the LLM. Thank you very much for going into so much detail and taking the time to explain it to me, I really appreciate it ^^

**Reply by angelsenvy228** [reply]
Seems to me there is no reason why you cannot set the program to, after a certain interval, pick a topic with some relation to the previous conversation inputs, or some degree of randomness, and have the AI say it. to say you have to wait for true AGI and it's impossible to do with current AI tech is unrealistic. the model is still being given a "prompt" it is just automatically generated and is something like "say something to continue the conversation" and doesn't print in the chat log? I've watched AI continue talking and prompting seemingly on their own, however it was accomplished it doesn't seem impossible.
